src/elements/EditableTable.tsx

```diff
@@ -0,0 +1,303 @@
+import Box, { BoxProps } from "@vertigis/web/ui/Box";
+import Edit from "@vertigis/web/ui/icons/Edit";
+import { FormElementProps, FormElementRegistration } from "@geocortex/workflow/runtime";
+import IconButton, { IconButtonProps } from "@vertigis/web/ui/IconButton";
+import Input from "@vertigis/web/ui/Input";
+import React, { useState, useEffect, CSSProperties } from "react";
+import Save from "@vertigis/web/ui/icons/Save";
+import Table, { TableProps } from "@vertigis/web/ui/Table";
+import TableBody from "@vertigis/web/ui/TableBody";
+import TableCell, { TableCellProps } from "@vertigis/web/ui/TableCell";
+import TableFooter from "@vertigis/web/ui/TableFooter";
+import TableHead from "@vertigis/web/ui/TableHead";
+import TableRow from "@vertigis/web/ui/TableRow";
+import DynamicIcon from "@vertigis/web/ui/DynamicIcon/DynamicIcon";
+
+type Total = Record<string, any>;
+type Row = Record<string, any>;
+type SettableBoxProps = Pick<BoxProps, "maxHeight" | "maxWidth">;
+type SettableTableProps = Pick<TableProps, "size" | "stickyHeader">;
+
+interface Column {
+    alias?: string;
+    align?: TableCellProps["align"];
+    format?: (value: any) => any;
+    parse?: (
+        event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+        row: Row,
+        column: Column
+    ) => any;
+    includeTotal?: boolean;
+    editable: boolean;
+    type: string;
+    name: string;
+}
+
+interface EditableTableElementProps
+    extends FormElementProps<Row[]>,
+        SettableBoxProps,
+        SettableTableProps {
+    cols: Column[];
+    rows: Row[];
+    editTooltip?: string;
+    saveTooltip?: string;
+    includeTotals?: boolean;
+    footerLabel?: string;
+    footerStyle?: CSSProperties;
```

Comment:
We want to style the footer is all.  I guess they could do that through an inject css.  We definitely want the footer to be styled differently from the body to easily identify a total.  The other option is to have a default footer style that's baked into the element.

Reply:
Let's just figure out what a table footer ought to look like in Web and style it that way directly in the component. 

Inject CSS is strongly discouraged in Web. None of the class names are predictable and the internal structure of the MUI components is always changing.

Reply:
Basically trying to style the footer.  I'm trying to avoid conflicting with Web's styling but I guess it's just making it someone else's problem (and potentially ours).  I'm open to ideas but I believe it's important to differentiate the footer from the body.  We could just have a style in the element...

Reply:
Fair enough. I think as long as we keep it super minimal and rely on the same variables that the header uses it will be fine. We just need to avoid exact sizes/fonts/colors/etc. See my other comment on the styles.